cmake/GitHubRelease.cmake

```diff
@@ -70,9 +70,9 @@ function(check_release_and_get_latest owner repo version os arch out_var)
 endfunction()
 
 function(get_best_slang_binary_release_url out_var)
-    if(CMAKE_SYSTEM_PROCESSOR MATCHES "x86_64|amd64")
+    if(CMAKE_SYSTEM_PROCESSOR MATCHES "x86_64|amd64|AMD64")
         set(arch "x86_64")
-    elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64|arm64")
+    elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64|ARM64|arm64")
```

Comment:
On Windows, it looks like the value of CMAKE_SYSTEM_PROCESSOR comes from an environment variable, "PROCESSOR_ARCHITECTURE".
https://cmake.org/cmake/help/latest/variable/CMAKE_HOST_SYSTEM_PROCESSOR.html#variable:CMAKE_HOST_SYSTEM_PROCESSOR

And it looks like the possible values are AMD64, IA64, ARM64 or x86:
https://learn.microsoft.com/en-us/windows/win32/winprog64/wow64-implementation-details?redirectedfrom=MSDN
Note that ARM64 should be in capital letters.
I am not sure if x86_64 is correct or not; it may need to be x86.

Reply:
we don't make releases for x86 so it would always fail anyway.